pypsa/optimization/constraints.py

```diff
@@ -52,29 +53,35 @@ def define_operational_constraints_for_non_extendables(
         name of the network component
     attr : str
         name of the attribute, e.g. 'p'
+    transmission_losses : int
+        Whether to consider transmission losses
     """
-    dispatch_lower: DataArray | tuple
-    dispatch_upper: DataArray | tuple
-
-    fix_i = n.get_non_extendable_i(c)
-    fix_i = fix_i.difference(n.get_committable_i(c)).rename(fix_i.name)
+    component = as_components(n, c)
+    fix_i = component.fixed
+    fix_i = fix_i.difference(component.committables).rename(f"{fix_i.name}-fix")
 
     if fix_i.empty:
         return
 
-    nominal_fix = n.static(c)[nominal_attrs[c]].reindex(fix_i)
-    min_pu, max_pu = get_bounds_pu(n, c, sns, fix_i, attr)
-    lower = min_pu.mul(nominal_fix)
-    upper = max_pu.mul(nominal_fix)
+    nominal_fix = component.as_xarray(component.nominal_attr).rename(
+        {component.name: fix_i.name}
+    )
```

Comment:
```suggestion
    nominal_fix = component.as_xarray(component.nominal_attr, inds=fix_i)
```
this already renames the dimension

Reply:
ok

Reply:
Just a note here that this will rename the dimension to whatever `fix_i` had as an index name. But we want to be stricter here and have fixed dimension names, so always return the same, no matter what your passed selection index is